### ISSUE TITLE
What's wrong with using context processors?

### ISSUE THREAD

**Issue by x-yuri** [niwinz/django-jinja#235]
Well, I've just found [the issue][1] that led to it. @aaugustin Care to join the discussion? I don't see any strong arguments about context processors being a bad design decision. Not until we know how much overhead using them adds.

The way the documentation states it, it's not clear why they are not recommended. I thought there's some bug, that they don't work under some conditions. That led me to "Can I at least use the `request` object?" It's [`django-jinja`][2] that adds it, but maybe passing `request` to templates is flawed in one way or another. And it's going to be removed in the near future (obsolete).

For now, I can't find a better phrasing than:

> `jinja` comes with notions of constants and globals. You might want to use them in place of context processors that don't need the `request` object.

In place of:

> The usage of context processors is not the recommended way anymore, and with django-jinja you can done it setting global data or global constants. See below, in the django 1.8 configuration related section.

Also,

> It is a helper to use django’s context processors with jinja2 backend for django 1.8.

Did you possibly mean `django >= 1.8`?

> Furthermore, it is now not the recommended way to setup variables in the context and the purpose of its existence is a help for migrations.

I'd drop this line.

[1]: https://github.com/niwinz/django-jinja/issues/91
[2]: https://github.com/niwinz/django-jinja/blob/2.4.1/django_jinja/backend.py#L79

**Comment by aaugustin** [comment]
I made my arguments in #91. If you disagree, that's fine :-)

**Comment by x-yuri** [comment]
Okay, but the way the documentation conveys it is confusing. We need to come up with better phrasing. The way I understand your point:

```diff
-The usage of context processors is not the recommended way anymore, and with django-jinja
-you can done it setting global data or global constants. See below, in the django 1.8
-configuration related section.
+The use of context processors is discouraged. Because they don't always need the `request`
+object, and many return a constant. You can use `globals`, and `constants` for that
+purpose.
```

Or we can not discourage, but point out that there's an alternative to context processors, as in the the message above (original post).

**Comment by DylanYoung** [comment]
I have to admit that I'm confused by this as well, both here and in the official django docs.

As far as i can see there is currently no substitute for variables that are tied to the lifecycle of the request (aside from some sort of caching of the result).  Django docs suggest similarly:

> The original use case for adding context processors for Jinja2 involved:
>Making an expensive computation that depends on the request.
>Needing the result in every template.
>Using the result multiple times in each template.
>Unless all of these conditions are met, passing a function to the template is simpler and more in line with the design of Jinja2.  

It seems to me that this is far too strict and the case where we actually need the context processors boils down to "the computation is expensive and used in multiple templates" (in the single template case, a "set" call can be used).

**Comment by DylanYoung** [comment]
Also, does this package still provide anything that Django doesn't?

**Comment by wizpig64** [comment]
marking this as closed as #261 is a duplicate of this.
